sensor/common/collector/service_impl.go

```diff
@@ -0,0 +1,77 @@
+package collector
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/logging"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+const maxBufferSize = 10000
+
+type Option func(*serviceImpl)
+
+// WithAuthFuncOverride sets the AuthFuncOverride.
+func WithAuthFuncOverride(overrideFn func(context.Context, string) (context.Context, error)) Option {
+	return func(srv *serviceImpl) {
+		srv.authFuncOverride = overrideFn
+	}
+}
+
+type serviceImpl struct {
+	sensor.UnimplementedCollectorServiceServer
+
+	queue chan *sensor.ProcessSignal
+
+	authFuncOverride func(context.Context, string) (context.Context, error)
+}
+
+func newService() Service {
+	return &serviceImpl{
+		queue: make(chan *sensor.ProcessSignal, maxBufferSize),
```

Comment:
We should add metrics to this buffer. Also, consider using `ScaleSizeOnNonDefault` and an environment variable to configure this max size.